### PR TITLE
Multi-threads: add scale number

### PR DESCRIPTION
**Pull request by ebattat**
Fixes:

Create a separate render variable to scale number for update it in state_signals_exporter_pod.yaml

### CODE REVIEW

File: benchmark_runner/common/template_operations/template_operations.py [hunk 1]
```diff
@@ -94,12 +94,15 @@ def __build_template_data(self, previous_data: dict, template_root_data: dict):
                 **extra_data, **extra_kind_data, **extra_runtype_data, **extra_kind_runtype_data}
 
     @logger_time_stamp
-    def generate_yamls_internal(self, scale: str = None, scale_node: str = None, redis: str = None):
+    def generate_yamls_internal(self, scale: str = None, scale_num: str = None, scale_node: str = None, redis: str = None):
```
Comment (RobertKrawitz) [inline benchmark_runner/common/template_operations/template_operations.py:97]: It looks like `scale` here is really an index, and `scale_num` is the total scale.  I suggest naming these parameters `scale_idx` and `scale` or similar.

These should be passed in as ints, and this routine should convert them to strings.  These are conceptually numbers, but they need to be converted into strings for template purposes.  That should be done below.

File: benchmark_runner/common/template_operations/template_operations.py [hunk 2]
```diff
@@ -125,6 +128,7 @@ def generate_yamls_internal(self, scale: str = None, scale_node: str = None, red
             'standard_output_file': self.__standard_output_file,
             'standard_template_file': self.__standard_template_file,
             'scale': scale,
+            'scale_num': scale_num,
```
Comment (RobertKrawitz) [inline benchmark_runner/common/template_operations/template_operations.py:131]: This is where `scale` and `scale_num` or whatever you want to call it should be converted from numbers to strings.

File: benchmark_runner/common/template_operations/template_operations.py [hunk 3]
```diff
@@ -181,7 +183,7 @@ def generate_yamls(self, scale: str = '', scale_nodes: list = [], redis: str = N
             if scale_nodes:
                 for bulk in bulks:
                     for num in bulk:
-                        p = Process(target=self.generate_yamls_internal, args=(str(num), scale_nodes[int(num/scale)], redis,))
+                        p = Process(target=self.generate_yamls_internal, args=(str(num), str(scale * len(scale_nodes)), scale_nodes[int(num/scale)], redis,))
```
Comment (RobertKrawitz) [inline benchmark_runner/common/template_operations/template_operations.py:186]: Again, just pass num and scale as integers, and let `generate_yamls_internal` convert them to strings.

**Review by RobertKrawitz — CHANGES_REQUESTED** [review]

File: benchmark_runner/common/template_operations/template_operations.py (hunk 3 above)
Comment (ebattat) [reply benchmark_runner/common/template_operations/template_operations.py:186]: Not see it as problem

File: benchmark_runner/common/template_operations/template_operations.py (hunk 1 above)
Comment (RobertKrawitz) [reply benchmark_runner/common/template_operations/template_operations.py:97]: Casts should happen as close to the point of use as possible.  The caller shouldn't know the use to which the data is being put.  What if the callee needs to use it both ways?

Comment (ebattat) [reply benchmark_runner/common/template_operations/template_operations.py:97]: In this way, I cast it once and not several times

Comment (RobertKrawitz) [reply benchmark_runner/common/template_operations/template_operations.py:97]: Why does that matter?  This is not extremely performance-sensitive code where the cast value needs to be cached.

Comment (RobertKrawitz) [reply benchmark_runner/common/template_operations/template_operations.py:97]: If you need to cache something for performance reasons, that's one thing.  If not, it's a lot easier to read and maintain if it's passed around in its conceptual form and only cast at point of use.

File: benchmark_runner/common/template_operations/template_operations.py [hunk 4]
```diff
@@ -94,12 +94,15 @@ def __build_template_data(self, previous_data: dict, template_root_data: dict):
                 **extra_data, **extra_kind_data, **extra_runtype_data, **extra_kind_runtype_data}
 
     @logger_time_stamp
-    def generate_yamls_internal(self, scale: str = None, scale_node: str = None, redis: str = None):
+    def __generate_yamls_internal(self, scale: str = None, scale_num: str = None, scale_node: str = None, redis: str = None):
```
Comment (RobertKrawitz) [inline benchmark_runner/common/template_operations/template_operations.py:97]: I still don't like casting it outside, but marking this function private guarantees that it is not an API that somebody else can use.

**Review by RobertKrawitz — APPROVED** [review]

**Pull request merged**